### PR TITLE
Add reviewers and reviewer selection to response detail page

### PR DESCRIPTION
**Pull request by greeffjean**
Updated Test issues for Response List and Response Page, 
All requirements for iss799 brief have been completed

### CODE REVIEW

**Review by avishkar58 — CHANGES_REQUESTED** [review]
Hey Jean,
I really like the UI here, it looks really nice! Good job!
I left a few comments on improving the code quality. Also note that the "New Tag" functionality is broken, I suspect this happened when you were resolving some merge conflicts. Some other things I picked up while testing this:
- It allows me to assign the same reviewer multiple times, please add some validation to prevent this
- As I assign and remove reviewers, the number of "Assign Reviewer" buttons keeps increasing. There should only be a number of slots corresponding to how many came through from the responseDetail API. So, if there were 3 slots there, and all of them are filled with reviewers, there should no *no* Assign Reviewer buttons left.

**Comment by avishkar58** [comment]
There are a lot of warnings coming from the ResponsePage.js. Please check  and fix these (they are visible in the console and in the docker terminal)

File: webapp/src/pages/ResponseList/components/ResponseListForm.js [hunk 1]
```diff
@@ -36,10 +36,11 @@ class ResponseListForm extends Component {
             responsesService.getResponseList(this.props.event.id, false, []),
             applicationFormService.getQuestionList(this.props.event.id)
         ]).then(responses => {
+            console.log(responses)
             this.setState({
                 tags: responses[0].tags,
                 responses: responses[1].responses,
-                questions: responses[2].questions.filter(q => q.type !== "information"),
+                questions: responses[2].questions ? responses[2].questions.filter(q => q.type !== "information") : null,
```
[the remark this answers is not in the document]
Comment (greeffjean) [reply webapp/src/pages/ResponseList/components/ResponseListForm.js:43]: 👍

File: webapp/src/pages/ResponsePage/ResponsePage.js [hunk 2]
```diff
@@ -256,54 +176,315 @@ class ResponsePage extends Component {
         }
     }
 
+    // render Delete Modal
+    renderDeleteModal() {
```
[the remark this answers is not in the document]
Comment (greeffjean) [reply webapp/src/pages/ResponsePage/ResponsePage.js:180]: Thanks for this, I see your point

File: webapp/src/pages/ResponsePage/ResponsePage.js [hunk 3]
```diff
@@ -256,54 +176,315 @@ class ResponsePage extends Component {
...
     renderTagModal() {
         const { eventLanguages } = this.state;
+
         if (eventLanguages) {
             return <TagModal
+                keys={this.state.keys}
+                i18nt={this.props.i18n}
                 t={this.props.t}
-                postTag={this.addTag}
+                postTag={(tags) => this.postTag(tags, "tagList")}
```
[the remark this answers is not in the document]
Comment (greeffjean) [reply webapp/src/pages/ResponsePage/ResponsePage.js:330]: I think this was a merge confusion, thanks for the last bit of advice noted!

File: webapp/src/pages/ResponsePage/ResponsePage.js [hunk 4]
```diff
@@ -256,54 +176,315 @@ class ResponsePage extends Component {
...
+    // Remove Reviewer
+    removeReview(type, reviewer_user_id) {
+        this.setState({
+            removeModalVisible: { boolean: true, type: "review", itemId: reviewer_user_id },
+            // tagToRemove: tag_id
+        });
+    }
+
+    removeReviewerService() {
+        const { applicationData, removeModalVisible } = this.state;
```
[the remark this answers is not in the document]
Comment (greeffjean) [reply webapp/src/pages/ResponsePage/ResponsePage.js:399]: Thanks for this again, I see now where I went wrong, I tried to simplify and reduce the amount of code but I can see I was lacking consideration for the reader and I think its better to write clear code even if it means more code

**Comment by greeffjean** [comment]
In the scenario where a user will delete an assigned reviewer, and activate thereafter the reviewerModal, would you want to deleted reviewer to be back on the list? 

Also Im handling all the delete, post and listing requests locally in the state, is this fine or would you rather the functionality rely more on the API responses?

**Comment by greeffjean** [comment]
> Hey Jean,
> I really like the UI here, it looks really nice! Good job!
> I left a few comments on improving the code quality. Also note that the "New Tag" functionality is broken, I suspect this happened when you were resolving some merge conflicts. Some other things I picked up while testing this:
> 
> * It allows me to assign the same reviewer multiple times, please add some validation to prevent this
> * As I assign and remove reviewers, the number of "Assign Reviewer" buttons keeps increasing. There should only be a number of slots corresponding to how many came through from the responseDetail API. So, if there were 3 slots there, and all of them are filled with reviewers, there should no Assign Reviewer buttons left.

Just Check the addTag Service, there is a error that's being thrown, other than that the functionality should be working as usual.

File: webapp/src/pages/ResponsePage/ResponsePage.js [hunk 5]
```diff
@@ -253,61 +175,334 @@ class ResponsePage extends Component {
...
+                                </button>
+                        </div>
+                    }
+                    else {
+                        return <div className="reviewer">
+                            <label>{"Reviewer" + num}</label>
+                            <div>
+                                <p>{val.user_title} {val.firstname} {val.lastname}</p>
+                                
+                                {val.completed ? <p className="review-completed">Completed</p>
```
Comment (avishkar58) [inline webapp/src/pages/ResponsePage/ResponsePage.js:356]: Please translate the text in this section ("Assign Reviewer", "Completed" and "Incomplete")

File: webapp/src/pages/ResponsePage/components/ReviewModal.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+                            <div className="modal-body">
+                                {reviewers && reviewers.map(val => {
+                                    return <button onClick={(e) => this.handleSelect(val)}
+                                        className={selectedReviews.includes(val) ? "review-select active" : "review-select"}
+                                        key={val.reviewer_user_id}
+                                    >
+                                        <label> {val.user_title} {val.firstname} {val.lastname} </label>
+                                        
+                                        <div>
+                                        <p>Reviews Allocated: {val.reviews_allocated} </p>
```
Comment (avishkar58) [inline webapp/src/pages/ResponsePage/components/ReviewModal.js:57]: Please translate the text here.

**Review by avishkar58 — CHANGES_REQUESTED** [review]
Thanks Jean, the code looks much better! Will test it again tonight

**Comment by avishkar58** [comment]
> In the scenario where a user will delete an assigned reviewer, and activate thereafter the reviewerModal, would you want to deleted reviewer to be back on the list?
> 
Yes please

> Also Im handling all the delete, post and listing requests locally in the state, is this fine or would you rather the functionality rely more on the API responses?
It's generally always better to rely on the API response, that way you know for sure that the front-end is displaying data that is consistent with what's in the back-end (and that it will be the same next time the user visits the page).

**Comment by greeffjean** [comment]
> > In the scenario where a user will delete an assigned reviewer, and activate thereafter the reviewerModal, would you want to deleted reviewer to be back on the list?
> 
> Yes please
> 
> > Also Im handling all the delete, post and listing requests locally in the state, is this fine or would you rather the functionality rely more on the API responses?
> 
> It's generally always better to rely on the API response, that way you know for sure that the front-end is displaying data that is consistent with what's in the back-end (and that it will be the same next time the user visits the page).

I will make changes tomorrow

File: webapp/src/pages/ResponsePage/ResponsePage.css [hunk 7]
```diff
@@ -29,12 +25,19 @@ button {
...
- .table-wrapper .btn-primary {
-    background:hsl(208deg 84% 59%);
+label {
+    font-size: 12px;
+    color: grey;
+    margin: 0 1px;
+}
+
+
+.table-wrapper.response-page .btn-primary {
```
Comment (greeffjean) [inline webapp/src/pages/ResponsePage/ResponsePage.css:35]: ### Let's keep global styles in App.css and only if they've been tested across the entire app. This one in particular significantly impacts the readability of the application form (so please change to to be specific to this page)

I believe I have attributed this specifically to this page please see class "response-page" ?

Comment (avishkar58) [reply webapp/src/pages/ResponsePage/ResponsePage.css:35]: The comment was referring to the "label" one above, not this one...

**Comment by greeffjean** [comment]
Since the reviewsApi is not built out yet I can't change the way its integrated right now, just for interest sake if I post a tag will it update the response from:  `responsesService.getResponseDetail`?

File: webapp/src/pages/ResponsePage/ResponsePage.js (hunk 5 above)
Comment (avishkar58) [reply webapp/src/pages/ResponsePage/ResponsePage.js:356]: @greeffjean - please fix this.

**Review by avishkar58 — APPROVED** [review]

**Pull request merged**